presto-verifier/src/main/java/com/facebook/presto/verifier/Validator.java

```diff
@@ -238,6 +268,36 @@ else if (controlResult.getState() != State.SUCCESS) {
         return true;
     }
 
+    private QueryResult executeQueryTest()
```

Comment:
this won't run the teardown queries if the setup or test query fails, or if there's an exception. Put the tear down in a finally-block